omni-relayer/src/utils/near.rs

```diff
@@ -88,10 +88,51 @@ pub async fn handle_streamer_message(
                 )
                 .await;
             }
-            Nep141LockerEvent::FinTransferEvent { .. }
-            | Nep141LockerEvent::LogMetadataEvent { .. }
-            | Nep141LockerEvent::SignClaimNativeFeeEvent { .. }
-            | Nep141LockerEvent::ClaimFeeEvent { .. } => {}
+            Nep141LockerEvent::FinTransferEvent {
+                ref nonce,
+                ref transfer_message,
+            } => {
+                if nonce.is_none() {
+                    utils::redis::add_event(
+                        redis_connection,
+                        utils::redis::NEAR_FIN_TRANSFER_EVENTS,
+                        transfer_message.origin_nonce.0.to_string(),
+                        log,
+                    )
+                    .await;
+                }
+            }
+            Nep141LockerEvent::ClaimFeeEvent {
+                ref transfer_message,
+                ref native_fee_recipient,
+            } => {
+                if native_fee_recipient == &config.near.relayer_address_on_evm {
+                    utils::redis::add_event(
+                        redis_connection,
+                        utils::redis::NEAR_FIN_TRANSFER_EVENTS,
+                        transfer_message.origin_nonce.0.to_string(),
+                        log,
+                    )
+                    .await;
+                }
+            }
+            Nep141LockerEvent::SignClaimNativeFeeEvent {
+                ref claim_payload, ..
```

Comment:
Will this be parsed correctly?

https://github.com/Near-One/omni-bridge/blob/develop/near/nep141-locker/src/lib.rs#L364

Reply:
What do you mean? I don't see a problem here. Since we don't care about signature I just retrieved the payload